cluster-api/src/main/java/io/aiven/klaw/clusterapi/controller/ClusterApiController.java

```diff
@@ -197,8 +209,14 @@ public ResponseEntity<ApiResponse> createTopics(
       @RequestBody @Valid ClusterTopicRequest clusterTopicRequest) {
     try {
       log.info("createTopics clusterTopicRequest {}", clusterTopicRequest);
-      return new ResponseEntity<>(
-          apacheKafkaTopicService.createTopic(clusterTopicRequest), HttpStatus.OK);
+      if (AclsNativeType.CONFLUENT_CLOUD == clusterTopicRequest.getAclsNativeType()) {
+        return new ResponseEntity<>(
+            confluentCloudApiService.createTopic(clusterTopicRequest), HttpStatus.OK);
+      } else {
+        return new ResponseEntity<>(
+            apacheKafkaTopicService.createTopic(clusterTopicRequest), HttpStatus.OK);
+      }
```

Comment:
A Future Enhancement : If we implement an interface for the different flavours with the create/update/list etc in that interface we could then do one check at the start to initialize with the correct instance and then we would be able to remove any if else statements depending on cloud type.

Reply:
Yes, I thought about it last week, and it would be a good improvement. agree.